### PR TITLE
feat: updated bitcoin-contracts package and rpc params

### PR DESCRIPTION
**Pull request by Polybius93**
> Try out this version of Leather — [download extension builds](https://github.com/leather-wallet/extension/actions/runs/6955655653).

This PR introduces some improvements on the Bitcoin Contracts:

Updated @dlc-link/dlc-tools package.

Modified acceptBitcoinContract RPC message parameters:
- Removed attestorList (_unnecessary in the new design_).
- Added bitcoinNetwork to ensure wallet compatibility with the offer's network.

Instead of the Expiration Date, the request page will show the Emergency Refund Time of the DLC.

### CODE REVIEW

File: src/shared/rpc/methods/accept-bitcoin-contract.ts [hunk 1]
```diff
@@ -16,7 +18,7 @@ export enum BitcoinContractResponseStatus {
   SUCCESS = 'Accepting Bitcoin Contract offer was successful',
   BROADCAST_ERROR = 'There was an error while broadcasting the Bitcoin Contract transaction',
   INTERFACE_ERROR = 'There was an error while interacting with the Bitcoin Contract interface',
-  NETWORK_ERROR = "The wallet's current selected network is not supported",
+  NETWORK_ERROR = "The wallet's current selected network is not matching the one in the Bitcoin Contract offer",
```
Comment (kyranjamie) [inline src/shared/rpc/methods/accept-bitcoin-contract.ts]:
```suggestion
  NETWORK_ERROR = "The wallet's current network does not match the one in the Bitcoin Contract offer",
```

File: src/app/common/hooks/use-bitcoin-contracts.ts [hunk 2]
```diff
@@ -77,11 +77,19 @@ export function useBitcoinContracts() {
 
     if (!currentAddressPrivateKey) return;
 
+    let bitcoinContractStorageApiUrl: string;
+    if (currentBitcoinNetwork.chain.bitcoin.bitcoinNetwork === 'testnet') {
+      bitcoinContractStorageApiUrl = 'https://testnet.dlc.link/storage-api';
+    } else {
+      bitcoinContractStorageApiUrl = 'https://devnet.dlc.link/storage-api';
+    }
```
Comment (kyranjamie) [inline src/app/common/hooks/use-bitcoin-contracts.ts]: Less mutative alternative

```ts
const bitcoinContractStorageMap: Record<'testnet' | 'regtest', string> = {
  testnet: 'https://testnet.dlc.link/storage-api',
  regtest: 'https://devnet.dlc.link/storage-api'
};
const bitcoinContractStorageApiUrl = bitcoinContractStorageMap[currentBitcoinNetwork.chain.bitcoin.bitcoinNetwork];

**Review by kyranjamie — APPROVED** [review]
Thanks @Polybius93 

See comments, otherwise LGTM

File: src/shared/rpc/methods/accept-bitcoin-contract.ts (hunk 1 above)
Comment (Polybius93) [reply src/shared/rpc/methods/accept-bitcoin-contract.ts]: Yes, thank you, much better!

File: src/app/common/hooks/use-bitcoin-contracts.ts (hunk 2 above)
Comment (Polybius93) [reply src/app/common/hooks/use-bitcoin-contracts.ts]: Great! I will change it!

**Pull request merged**